### ISSUE TITLE
Is it possible to do nested joins?

### ISSUE THREAD

**Issue by zacharynevin** [Bill4Time/mongo-fast-join#6]
I have another question (sorry!) that has to do with joins on nested data. Consider an example where I can have events, and each event can have a number of members:

```
// event
{
    id: {Integer},
    title: {String},
    description: {String},
    start: {Date},
    end: {Date},
    owner: {
        id: {Integer}
    },
    participants: [{
        id: {Integer}
    },{
        id: {Integer}
    }]
}
```

The `participants.id` attribute is meant to be a reference to the `id` attribute in the `users` document store, below:

```
// users
{
    id: {Integer},
    fname: {String},
    lname: {String},
    gender: {String}
}
```

Is it possible to do a join such that I produce the following, below? Is it advisable?

```
// event
{
    id: {Integer},
    title: {String},
    description: {String},
    start: {Date},
    end: {Date},
    owner: {
        id: {Integer}
    },
    participants: [{
        id: {Integer},
        fname: {String},
        lname: {String},
        gender: {String}           
    },{
        id: {Integer},
        fname: {String},
        lname: {String},
        gender: {String}           
    }]
}
```

Thank you for your time.

**Comment by khous** [comment]
TLDR: No.

Hey! No worries asking questions. It's extremely gratifying to have my work considered as a possible solution to your problem. I haven't worked on this since the last commit so it takes me a moment to recollect the exact state of the project.

If you take a look at this version of the readme, particularly line 58:

https://github.com/Bill4Time/mongo-fast-join/commit/d0618c05359eb6d534b255dacde62196b5808a9c#diff-04c6e90faac2675aa89e2176d2eec7d8R58

you will find a more complete example of usage and an exact answer to your question.

Months ago I planned on adding this functionality but never got the opportunity to do so. I would accept a pull request adding it.

**Comment by zacharynevin** [comment]
Thank you! I will look through the code and see if I can work it out.

When I got rid of the ORM I underestimated the difficulty of impedance-mismatch. Not only this, but after going through all my data models I realised that the majority of them can be denormalised entirely without having data duplication issues. Most of what I do doesn't require joins, except things like the example above. I have taken the plunge into MongoDB.

**Comment by zacharynevin** [comment]
I created a pull request. I hope I understood the join process correctly. Tell me what you think.
